actors/user/src/test/java/org/sunbird/user/AddressManagementActorTest.java

```diff
@@ -0,0 +1,138 @@
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.sunbird.cassandraimpl.CassandraOperationImpl;
+import org.sunbird.common.ElasticSearchUtil;
+import org.sunbird.common.models.response.Response;
+import org.sunbird.common.models.util.JsonKey;
+import org.sunbird.common.models.util.datasecurity.EncryptionService;
+import org.sunbird.common.request.Request;
+import org.sunbird.helper.ServiceFactory;
+import org.sunbird.user.actors.AddressManagementActor;
+import org.sunbird.user.util.UserActorOperations;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({
+  ServiceFactory.class,
+  ElasticSearchUtil.class,
+  EncryptionService.class,
+  org.sunbird.common.models.util.datasecurity.impl.ServiceFactory.class
+})
+@PowerMockIgnore({"javax.management.*"})
+public class AddressManagementActorTest {
+
+  private static final ActorSystem system = ActorSystem.create("system");
+  private static final Props props = Props.create(AddressManagementActor.class);
+  private static final CassandraOperationImpl cassandraOperation =
+      mock(CassandraOperationImpl.class);;
+
+  @Before
+  public void beforeEachTest() {
+
+    PowerMockito.mockStatic(ServiceFactory.class);
+    PowerMockito.mockStatic(org.sunbird.common.models.util.datasecurity.impl.ServiceFactory.class);
+    EncryptionService encryptionService = Mockito.mock(EncryptionService.class);
+    Mockito.when(
+            org.sunbird.common.models.util.datasecurity.impl.ServiceFactory
+                .getEncryptionServiceInstance(null))
+        .thenReturn(encryptionService);
+
+    try {
+      Mockito.when(encryptionService.encryptData(Mockito.anyString())).thenReturn("encrptUserId");
+    } catch (Exception e) {
+      fail("AddressManagementActorTest initialization failed");
+    }
+    when(ServiceFactory.getInstance()).thenReturn(cassandraOperation);
+    when(cassandraOperation.insertRecord(
+            Mockito.anyString(), Mockito.anyString(), Mockito.anyMap()))
+        .thenReturn(getSuccessResponse());
+    when(cassandraOperation.deleteRecord(
+            Mockito.anyString(), Mockito.anyString(), Mockito.anyString()))
+        .thenReturn(getSuccessResponse());
+  }
+
+  @Test
+  public void testInsertUserAddressSuccess() {
+    boolean result = testScenario(UserActorOperations.INSERT_USER_ADDRESS.getValue(), true);
+    assertTrue(result);
+  }
+
+  @Test
+  public void testUpdateAddressSuccess() {
+    boolean result = testScenario(UserActorOperations.UPDATE_USER_ADDRESS.getValue(), true);
+    assertTrue(result);
+  }
+
+  @Test
+  public void testInsertUserAddressFailureWithoutReqParams() {
+    boolean result = testScenario(UserActorOperations.INSERT_USER_ADDRESS.getValue(), false);
+    assertTrue(result);
+  }
+
+  @Test
+  public void testUpdateUserAddressFailureWithoutMandatoryFields() {
+    boolean result = testScenario(UserActorOperations.UPDATE_USER_ADDRESS.getValue(), false);
```

Comment:
why update fail for mandatory fields? does it expect whole current object state?

Reply:
delete address false.  & 1. containing id, 2. not containing id are 2 other test-cases